src/ReaderScreen/components/AudioPlayer/components/AudioControlBar.jsx

```diff
@@ -0,0 +1,216 @@
+import React, { useState, useEffect } from "react";
+import { View, Text, Pressable, Animated, Platform } from "react-native";
+import { Slider } from "@miblanchard/react-native-slider";
+import { BlurView } from "@react-native-community/blur";
+import PropTypes from "prop-types";
+import useTheme from "@common/context";
+import useThemedStyles from "@common/hooks/useThemedStyles";
+import { MusicNoteIcon, SettingsIcon, CloseIcon, PlayIcon, PauseIcon } from "@common/icons";
+import { STRINGS } from "@common";
+import { useAnimation, useDownloadManager, useAudioManifest } from "../hooks";
+import { audioControlBarStyles } from "../style";
+import ActionComponents from "./ActionComponent";
+import AudioSettingsModal from "./AudioSettingsModal";
+import DownloadBadge from "./DownloadBadge";
+import ScrollViewComponent from "./ScrollViewComponent";
+
+const AudioControlBar = ({
+  isPlaying,
+  handlePlayPause,
+  progress,
+  handleSeek,
+  isAudioEnabled,
+  handleTrackSelect,
+  onCloseTrackModal,
+  baniID,
+}) => {
+  const { theme } = useTheme();
+  const styles = useThemedStyles(audioControlBarStyles);
+  const [isSettingsModalOpen, setIsSettingsModalOpen] = useState(false);
+  const [isMoreTracksModalOpen, setIsMoreTracksModalOpen] = useState(false);
+  const { modalHeight, modalOpacity } = useAnimation(isSettingsModalOpen, isMoreTracksModalOpen);
+  const { tracks, currentPlaying, addTrackToManifest, removeTrackFromManifest, isTrackDownloaded } =
+    useAudioManifest(baniID);
+  const { isDownloading, isDownloaded, handleDownload } = useDownloadManager(
+    currentPlaying,
+    addTrackToManifest,
+    removeTrackFromManifest,
+    isTrackDownloaded
+  );
+
+  const formatTime = (seconds) => {
+    const mins = Math.floor(seconds / 60);
+    const secs = Math.floor(seconds % 60);
+    return `${mins}:${secs.toString().padStart(2, "0")}`;
+  };
+
+  // Calculate slider colors to avoid nested ternary
+  const getSliderMinTrackColor = () => {
```

Comment:
This function could be memoized with useMemo to avoid recalculating on every render.